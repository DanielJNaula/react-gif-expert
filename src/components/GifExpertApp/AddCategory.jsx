import { useState } from "react";
import PropTypes from 'prop-Types';

export const Addcategory = ( { onNewCategory }) => {

  const [inputValue, setInputValue] = useState('');
  const handleOnChangeInput = (event) => {
    setInputValue(event.target.value);
  };

  const handleOnSubmit = (event) =>{
    event.preventDefault();
    const newInputValue = inputValue.trim();
    
    if (newInputValue.length <= 1) return;

    //setCategories( categories => [...categories, inputValue])
    onNewCategory(newInputValue);
    setInputValue('');
  };

  return (
    <form onSubmit={ (event) => handleOnSubmit(event) } aria-label="form">
      <input 
        type="text" 
        placeholder="Buscar Gif"
        value={ inputValue }
        onChange={ (event) => handleOnChangeInput(event) }
      />
    </form>
  )
};

Addcategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired,
};
