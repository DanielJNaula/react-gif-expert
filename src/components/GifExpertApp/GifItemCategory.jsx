import PropTypes from 'prop-types'

export const GifItemCategory = ( { title, url, id } ) => {

  return (
    <div className="card">
      <img src={ url } alt={ title } />
      <p>{ title }</p>
    </div>
  )
}

GifItemCategory.propTypes = {
  title:PropTypes.string.isRequired,
  url:PropTypes.string.isRequired
};
