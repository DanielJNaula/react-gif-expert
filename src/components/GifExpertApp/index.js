
export * from './AddCategory'
export * from './GifGridCategory'
export * from './GifItemCategory'
export * from './ListCategories'