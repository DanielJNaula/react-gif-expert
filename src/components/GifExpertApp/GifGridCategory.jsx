import PropTypes from 'prop-types';
import { GifItemCategory } from "./GifItemCategory";
import { useFetchGifs } from "../../hooks/useFetchGifs";

export const GifGridCategory = ({ category }) => {
  const { gifs, isLoading} = useFetchGifs( category );
  
  return (
    <div>
      <h3>{ category }</h3>
      {
        isLoading && <h2 >Cargando....</h2>
      }
      <div className="card-grid">
        {
          gifs.map((image)=> (
            <GifItemCategory
              key={ image.id }
              { ...image }
            />
          ))
        }
      </div>
    </div>
  )
};

GifGridCategory.propTypes = {
  category: PropTypes.string.isRequired,
};
