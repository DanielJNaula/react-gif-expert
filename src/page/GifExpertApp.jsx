import { useState } from "react"
import { Addcategory, ListCategories } from "../components/GifExpertApp";

export const GifExpertApp = () => {
  const [categories, setCategories] = useState(['One Punch'])
 
  const handleAddCategory = ( newCategory ) => {
    if( categories.includes(newCategory) ) return;
    setCategories( [newCategory, ...categories] );
  }

  return (
    <>
      <h1>GifExpertApp</h1>
      <Addcategory onNewCategory= { (valueInput) => handleAddCategory(valueInput) } />
      <ListCategories categories={ categories } />
    </>
  )
}
