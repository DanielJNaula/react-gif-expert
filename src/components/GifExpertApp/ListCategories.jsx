import { GifGridCategory } from "./GifGridCategory"

export const ListCategories = ({ categories }) => {
  return (
      categories.map((category) => <GifGridCategory key={ category } category={ category }/>) 
  )
}
