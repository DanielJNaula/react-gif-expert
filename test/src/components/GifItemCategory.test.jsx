import { render, screen } from "@testing-library/react";
import { GifItemCategory } from "../../../src/components/GifExpertApp/GifItemCategory";

describe('Pruebas en componente <GifItemCategory />', () => {
  const title = 'Dragon Ball';
  const url = 'https://gifs/algo';

  test('El componente debe hacer snapshot', () => {
    const {container} = render(<GifItemCategory title={ title } url={ url }/>);

    expect(container).toMatchSnapshot();
  });

  test('Debe mostrar la imagen con el URL y el alt indicado', () => {
    render(<GifItemCategory title={ title } url={ url }/>);
    const { src, alt } = screen.getByRole('img');
   
    expect(src).toBe(url);
    expect(alt).toBe(title);
  });

  test('Debe mostrar el titulo en el componente', () => {
    render(<GifItemCategory title={ title } url={ url }/>);

    expect( screen.getByText( title ).innerHTML ).toBe( title );
  });
});