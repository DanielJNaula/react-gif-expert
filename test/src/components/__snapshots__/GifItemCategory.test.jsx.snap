// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente <GifItemCategory /> El componente debe hacer snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Dragon Ball"
      src="https://gifs/algo"
    />
    <p>
      Dragon Ball
    </p>
  </div>
</div>
`;
